### DISCUSSION TITLE
Does apache superset Oauth support 3 legged authentication

### DISCUSSION THREAD

**Discussion by Vamsy26** [apache/superset#21675] [Q&A / Help]
Hello All,

We are trying to integrate OAUTH2 with Apache superset with OIAM being the identity provider. Our OIAM is configured for 3 legged authentication. We are unable to authenticate or reach the identity provider as the session is getting expired. 

Reason: it says that the identity provider has a domain and superset is unable to parse the domain from its query sting parameters. 

question: 

1) Does apache superset supports 3 legged authentication? if yes, request to pls share the working configuration which parses the domain in query sting parameters.

2) Can we explicitly parse the query sting parameters which has domain in it?

**Comment by kamalkeshavani-aiinside** [comment]
Superset is basically using Flask app for authentication. You can try checking their [documentation](https://flask-appbuilder.readthedocs.io/en/latest/).
@dpgaspar any suggestions?
